specs/design-black-box.md

```diff
@@ -72,10 +72,121 @@ specific ConfigMaps/Secrets.
 
 ## Design Details
 
-### Public API
-
-To be documented during POC implementation.
-
+The proposed design is based on the PoC that contains draft implementation and provides the base idea of a target design. PoC is available in
+[argo-cd#fargo/engine](https://github.com/alexmt/argo-cd/tree/fargo/engine) and [flux#fargo](https://github.com/alexmt/flux/tree/fargo) 
+
+The GitOps engine API consists of three main packages:
+* The `utils` package. It consist of loosely coupled packages that implements K8S resource diffing, health assessment, etc
+* The `app` package that Application data structure definition and CRUD client interface.
```

Comment:
What would be the (breaking) impact for Argo CD to change this into `repo` / `Repository`? I think we already discussed this multiple times, but when we detach this engine from Argo the 'application' context does not make much sense anymore, as the engine itself and Flux, unlike Argo, are focused around working with repositories.

Reply:
+1

Reply:
I think you are right. After abstracting the manifest generation most of the Application spec is not used by the engine. Now it is much easier to come up with the Golang interface and remove the dependency on the Application.

Going through the app controller code and trying to find which methods should be exposed by the interface.  

I would like to name the interface `DeployableUnit`. This seems neutral to both Argo CD and Flux. Does the name sounds good to you? @hiddeco , @2opremio

Reply:
Whenever I talk about the action Flux performs with regard to cluster operations I talk about syncing (and updating image tag references), so for me `Syncable<Something>` would make more sense to me.

Reply:
My vote was `ManagedRepository` :)

Reply:
I guess Application is very Argo specific and Repo is very Flux specific :) 


How about `SyncUnit`? Both Argo CD and Flux uses term `syncing` so `SyncUnit` would be understandable to users from both communities.

Reply:
Worked on draft changes in the engine and definitely looks better if we remove reference to the application. Updated design doc and removed Application from it.


I've tried using different names that include `Source` ( `SyncSource`, `TruthSource`  ) but It looks strange that Source has Source and Destination. 
e.g.

```
mySource := SyncSource{}
mySource.Source = ...
mySource.Destination = ...
```

So temporary using `SyncUnit` and totally open to change it

Reply:
Replace `SyncUnit` with `Sync<Term>`. @hiddeco will setup a poll in slack to decide the best name :)